cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/model/ClusterModelStats.java

```diff
@@ -257,42 +248,48 @@ public String toString() {
 
   /**
    * Generate statistics of utilization for resources among alive brokers in the given cluster.
+   * Average and standard deviation calculations are based on brokers not excluded for replica moves.
    *
    * @param clusterModel The state of the cluster.
+   * @param optimizationOptions Options to take into account while retrieving cluster capacity.
    */
-  private void utilizationForResources(ClusterModel clusterModel) {
+  private void utilizationForResources(ClusterModel clusterModel, OptimizationOptions optimizationOptions) {
     // Average, maximum, and standard deviation of utilization by resource.
-    Map<Resource, Double> avgUtilizationByResource = new HashMap<>();
-    Map<Resource, Double> maxUtilizationByResource = new HashMap<>();
-    Map<Resource, Double> minUtilizationByResource = new HashMap<>();
-    Map<Resource, Double> stDevUtilizationByResource = new HashMap<>();
+    Map<Resource, Double> avgUtilizationByResource = new HashMap<>(Resource.cachedValues().size());
```

Comment:
Nit: I'd do:
```
final int resourceCachedValueSize = Resource.cachedValues().size();
Map<Resource, Double> maxUtilizationByResource = new HashMap<>(resourceCachedValueSize);
...
```

Reply:
I'd prefer not using the proposed approach -- please see my answer on when I think we should use the `final` keyword.